lib/remote/apilistener.cpp

```diff
@@ -1434,10 +1434,16 @@ void ApiListener::LogGlobHandler(std::vector<int>& files, const String& file)
 void ApiListener::ReplayLog(const JsonRpcConnection::Ptr& client)
 {
 	Endpoint::Ptr endpoint = client->GetEndpoint();
+	Defer resetEndpointSyncing ([this, &endpoint]() {
+		ObjectLock olock(endpoint);
+		endpoint->SetSyncing(false);
+
+		if (!m_LogFile) {
+			OpenLogFile();
+		}
```

Comment:
This now violates the following precondition of `OpenLogFile()`:
https://github.com/Icinga/icinga2/blob/e936c43e898efe0532da39c2362169748e993457/lib/remote/apilistener.cpp#L1362-L1363

So this has to happen within the loop while it's still holding that mutex.